### PR TITLE
[ContactStructuralMechanicsApplication] Transition PR. Making the search utility into process

### PR DESCRIPTION
**Pull request by loumalouomega**
This PR transforms the search utility into a process, this is done in order to be able to create a wrapper in order to simplify code and make it more user friendly. Additionally prepares the process for the Octree search coming soon.

### CODE REVIEW

File: applications/ContactStructuralMechanicsApplication/python_scripts/search_base_process.py [hunk 1]
```diff
@@ -224,7 +218,15 @@ def ExecuteFinalizeSolutionStep(self):
         Keyword arguments:
         self -- It signifies an instance of a class.
         """
-        pass
+        global_step = self.main_model_part.ProcessInfo[KM.STEP]
+        if self._get_if_is_interval():
+            modified = self.main_model_part.Is(KM.MODIFIED)
+            database_step_update = self.settings["search_parameters"]["database_step_update"].GetInt()
+            if modified is False and (self.database_step >= database_step_update or global_step == 1):
```
Comment (philbucher) [inline applications/ContactStructuralMechanicsApplication/python_scripts/search_base_process.py]:
```suggestion
            if not modified and (self.database_step >= database_step_update or global_step == 1):
```

File: applications/ContactStructuralMechanicsApplication/python_scripts/search_base_process.py [hunk 2]
```diff
@@ -240,15 +242,7 @@ def ExecuteAfterOutputStep(self):
...
-        global_step = self.main_model_part.ProcessInfo[KM.STEP]
-        if self._get_if_is_interval():
-            modified = self.main_model_part.Is(KM.MODIFIED)
-            database_step_update = self.settings["search_parameters"]["database_step_update"].GetInt()
-            if modified is False and (self.database_step >= database_step_update or global_step == 1):
-                for key in self.settings["search_model_part"].keys():
-                    if self.settings["search_model_part"][key].size() > 0:
-                        self.search_utility_list[key].ClearMortarConditions()
-                self.database_step = 0
+        pass
```
Comment (philbucher) [inline applications/ContactStructuralMechanicsApplication/python_scripts/search_base_process.py:245]: I would remove it if it does nothing

File: applications/ContactStructuralMechanicsApplication/python_scripts/search_base_process.py [hunk 3]
```diff
@@ -420,42 +415,9 @@ def _create_main_search(self, key = "0"):
...
-                    if simple_search:
-                        self.search_utility_list[key] = CSMA.SimpleContactSearch3D3N(self.computing_model_part, search_parameters)
-                    else:
-                        self.search_utility_list[key] = CSMA.AdvancedContactSearch3D3N(self.computing_model_part, search_parameters)
-                else:
-                    if simple_search:
-                        self.search_utility_list[key] = CSMA.SimpleContactSearch3D3N4N(self.computing_model_part, search_parameters)
-                    else:
-                        self.search_utility_list[key] = CSMA.AdvancedContactSearch3D3N4N(self.computing_model_part, search_parameters)
+        self.search_utility_list[key] = CSMA.ContactSearchProcess(self.computing_model_part, search_parameters)
```
Comment (philbucher) [inline applications/ContactStructuralMechanicsApplication/python_scripts/search_base_process.py:419]:
```suggestion
        self.search_utility_map[key] = CSMA.ContactSearchProcess(self.computing_model_part, search_parameters)
```

**Review by philbucher — DISMISSED** [review]
minor comment, consider them or not :)

File: applications/ContactStructuralMechanicsApplication/python_scripts/search_base_process.py (hunk 3 above)
Comment (loumalouomega) [reply applications/ContactStructuralMechanicsApplication/python_scripts/search_base_process.py:419]: Nope, changing this will break the code

**Comment by loumalouomega** [comment]
@philbucher I need reapproval

File: applications/ContactStructuralMechanicsApplication/python_scripts/search_base_process.py (hunk 3 above)
Comment (philbucher) [reply applications/ContactStructuralMechanicsApplication/python_scripts/search_base_process.py:419]: ok

**Review by philbucher — APPROVED** [review]

**Pull request merged**